README.md

```diff
@@ -198,114 +198,156 @@ Mulang is just a Haskell library. You can install it though cabal.
 
 But if you are not the Haskell inclined gal or guy - ok, I will try to forgive you - this code comes with a command line too. So you don't even have to typecheck!
 
-Sample CLI usage...
+## Sample CLI usage
 
-...with advanced expectations:
+### With advanced expectations:
 
 ```bash
 $ mulang '
 {
-   "expectations" : [
-      {
-         "negated" : false,
-         "subject" : [
-            "x"
-         ],
-         "transitive" : false,
-         "object" : {
-            "tag" : "Anyone",
-            "contents" : []
-         },
-         "tag" : "Advanced",
-         "verb" : "uses"
-      }
-   ],
-   "code" : {
-      "content" : "x = 1",
-      "language" : "Haskell"
+   "sample" : {
+      "language" : "Haskell",
+      "content" : "x = 1"
    },
-   "analyseSignatures" : false
-}' | json_pp
+   "spec" : {
+      "expectations" : [
+         {
+            "subject" : [
+               "x"
+            ],
+            "object" : {
+               "contents" : [],
+               "tag" : "Anyone"
+            },
+            "negated" : false,
+            "verb" : "uses",
+            "transitive" : false,
+            "tag" : "Advanced"
+         }
+      ],
+      "signatureAnalysisType" : {
+        "tag" : "NoSignatures",
+        "contents" : []
+      }
+   }
+}
+' | json_pp
 {
-   "results" : [
+   "signatures" : [],
+   "tag" : "AnalysisCompleted",
```

Comment:
Yes, changing it is a pain in the neck. There is little we can do about that. 

However, my next PR 's change `SumEncoding` strategies, so most of those problems nearly disappear